unifai/tools/tools.py

```diff
@@ -102,53 +102,162 @@ class FunctionName(Enum):
 class Tools:
     _api_key: str
     _api: ToolsAPI
+    _max_limit: int
 
     def __init__(self, api_key: str):
-        """
-        A class to interact with the Unifai Tools API.
-
-        :param api_key: the API key of your toolkit.
-        """
         self._api_key = api_key
         self._api = ToolsAPI(api_key)
         self.set_api_endpoint(BACKEND_API_ENDPOINT)
+        self._max_limit = 100 
 
     def set_api_endpoint(self, endpoint: str):
         self._api.set_endpoint(endpoint)
 
-    def get_tools(self, cache_control: bool = False) -> List[Dict[str, Any]]:
-        """
-        Get the list of tools in OpenAI API compatible format.
+    async def _fetch_static_tools(self, static_toolkits: List[str] = None, static_actions: List[str] = None) -> List[Dict[str, Any]]:
+        result_tools = []
+    
+        try:
+            if static_toolkits or static_actions:
+                actions = await self._api.fetch_static_resources(
+                    toolkit_ids=static_toolkits,
+                    action_ids=static_actions,
+                    limit=self._max_limit
+                )
+                
+                for action in actions:
+                    result_tools.append({
+                        "type": "function",
+                        "function": {
+                            "name": action.get("action", "unknown"),
+                            "description": action.get("description", ""),
+                            "parameters": {
+                                "type": "object",
+                                "properties": action.get("payload", {})
+                            }
+                        }
+                    })
+                
+                return result_tools
+        except Exception as e:
+            logger.warning(f"Failed to fetch resources in batch: {e}. Falling back to separate requests.")
+        
+        tasks = []
```

Comment:
there is no need for task at all, all actions can be fetched by a single api call